x-pack/plugins/fleet/public/applications/integrations/hooks/use_package_install.tsx

```diff
@@ -66,31 +74,50 @@ function usePackageInstall({
     [packages]
   );
 
+  const optionallyForceInstall = async (
+    installProps: InstallPackageProps,
+    prevStatus: PackageInstallItem
+  ) => {
+    const forceInstall = await confirmForceInstall(installProps);
+    if (forceInstall) {
+      installPackage({ ...installProps, force: true });
+    } else {
+      setPackageInstallStatus({ ...prevStatus, name: installProps.name });
+    }
+  };
+
   const installPackage = useCallback(
-    async ({
-      name,
-      version,
-      title,
-      fromUpdate = false,
-      isReinstall = false,
-    }: InstallPackageProps) => {
-      const currStatus = getPackageInstallStatus(name);
+    async (props: InstallPackageProps) => {
+      const {
+        name,
+        version,
+        title,
+        fromUpdate = false,
+        isReinstall = false,
+        force = false,
+      } = props;
+      const prevStatus = getPackageInstallStatus(name);
       const newStatus = {
-        ...currStatus,
+        ...prevStatus,
         name,
         status: isReinstall ? InstallStatus.reinstalling : InstallStatus.installing,
       };
       setPackageInstallStatus(newStatus);
 
-      const res = await sendInstallPackage(name, version, isReinstall);
+      const res = await sendInstallPackage(name, version, isReinstall || force);
       if (res.error) {
+        if (isVerificationError(res.error)) {
+          optionallyForceInstall(props, prevStatus);
```

Comment:
@nchaulet @kpollich I've moved to using overlays and a hook for the force install modal, I'm happy with how it's turned out it's a lot simpler.